src/cargo/core/resolver/version_prefs.rs

```diff
@@ -62,6 +68,9 @@ impl VersionPreferences {
                     .map(|deps| deps.iter().any(|d| d.matches_id(*pkg_id)))
                     .unwrap_or(false)
         };
+        if self.max_rust_version.is_some() {
+            summaries.retain(|s| s.rust_version() <= self.max_rust_version.as_ref());
```

Comment:
Should this still be called `sort_summaries` if it can remove versions?

Reply:
That name mismatch goes back to when I added `-Zdirect-minimal-versions`.

Have an idea for a better name?

Reply:
Not realy. Sorry.